src/wot.iced

```diff
@@ -0,0 +1,82 @@
+{Base} = require './base'
+{constants} = require './constants'
+{make_esc} = require 'iced-error'
+pgp_utils = require('pgp-utils')
+{json_stringify_sorted,unix_time,streq_secure} = pgp_utils.util
+schema = require './schema3'
+
+#==========================================================================
+
+exports.Attest = class Attest extends Base
+
+  constructor : (obj) ->
+    @wot = obj.wot
+    super obj
+
+  _v_stub_paths : () -> [ "body.wot_attest" ]
+  _type : () -> constants.sig_types.wot.attest
+
+  _type_v2 : (revoke_flag) ->
+    if @revoke? or revoke_flag then constants.sig_types_v2.wot.attest_with_revoke
+    else constants.sig_types_v2.wot.attest
+
+  _v_check : ({json}, cb) ->
+    obj = json.body.wot_attest
+
+    proof_schema = schema.dict({
+      check_data_json : schema.or([
+        schema.dict({name : schema.string(),   username : schema.string() }),
+        schema.dict({domain : schema.string(), protocol : schema.string() })
+      ])
+      state : schema.int()
+      proof_type : schema.int()
+    })
+
+    schm = schema.dict({
+      user : schema.dict({
+        username : schema.string()
+        uid : schema.uid().convert()
```

Comment:
do we need this? (if the goal is to save space)

Reply:
I guess we can do with one or the other...  Hmmm, but then which?